src/NUnitFramework/TestBuilder.cs

```diff
@@ -161,34 +133,45 @@ public static ITestResult RunTestCase(object fixture, string methodName)
             return RunTest(testMethod, fixture);
         }
 
-#if !NETSTANDARD1_3 && !NETSTANDARD1_6
-        public static ITestResult RunAsTestCase(Action action)
-        {
-            var method = action.Method;
-            var testMethod = MakeTestCase(method.DeclaringType, method.Name);
-            return RunTest(testMethod);
-        }
-#endif
-
         public static ITestResult RunTest(Test test)
         {
             return RunTest(test, null);
         }
 
         public static ITestResult RunTest(Test test, object testObject)
         {
-            return ExecuteWorkItem(PrepareWorkItem(test, testObject));
+            return ExecuteWorkItem(CreateWorkItem(test, testObject));
         }
 
-        // NOTE: The following two methods are separate in order to support
-        // tests that need to access the WorkItem before or after execution.
+        public static CompositeWorkItem CreateWorkItem(Type type)
+        {
+            return CreateWorkItem(MakeFixture(type)) as CompositeWorkItem;
```

Comment:
Are we expecting consuming code to be tolerant of null returns? I prefer direct casts when possible because diagnosis at the call site is easier.

Reply:
Different philosophy on my part for tests than for production code, I guess. In general, the return is used immediately and we get a null reference exception. I think it's probably six of one etc. unless the return is being held on to for a long time and then used later.

Reply:
I added a comment clarifying that the cast is guaranteed - WorkItem created from a TestSuite is always composite.

Reply:
In my mind, safe cast + "cast is guaranteed" comment can be refactored to direct cast and communicate precisely the same thing. But no big deal.

Reply:
Hmm... I don't think it communicates that. IT gives rise to the comment "How do you know this cast won't throw?" There's no satisfying everyone, I guess. 😄

Reply:
I changed the return type. Problem solved!

Reply:
In either case you'd also have to explain _how_ you know it won't throw. But no big deal; I'm satisfied either way. Just exchanging viewpoints!